sentry-core/src/main/java/io/sentry/core/ILogger.java

```diff
@@ -1,7 +1,42 @@
 package io.sentry.core;
 
+import io.sentry.core.util.Nullable;
+
 /** Sentry SDK internal logging interface. */
 public interface ILogger {
+
+  /**
+   * This method is supposed to be statically imported and provides a shortcut for logging the
+   * message only if logger is not null.
+   *
+   * @param logger the logger or null if none found
+   * @param level the log level
+   * @param message the message
+   * @param args the formatting arguments
+   * @see #log(SentryLevel, String, Object...)
+   */
+  static void log(@Nullable ILogger logger, SentryLevel level, String message, Object... args) {
+    if (logger != null) {
```

Comment:
Although not documented, it's nice to be able to swap loggers at runtime.
Would that work with?:

```java
ILogger currentLogger = logger;
if (currentLogger != null) {
  currentLogger.log(level, message, throwable);
}

Reply:
Yes, I understood that intention. That's why this is a static method that you pass a logger to. So that the next time, you're free to pass a different logger. That is the reason why the classes in transport don't hold on to the logger directly but rather hold the options instance and call this method as:

```java
log(options.getLogger(), ...);
```

so as long as there is some code holding the options instance able to set different loggers, this method support swapping the loggers.

Reply:
much nicer, thanks!